sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java

```diff
@@ -0,0 +1,605 @@
+            assertTrue(region.getBoundingBox().getWidth() <= result.getMetadata().getWidth() - region.getBoundingBox().getX());
+        }
+
+        // Make sure each bounding box is unique
+        for (int i = 0; i < listCropRegions.size(); i++) {
+            CropRegion region = listCropRegions.get(i);
+            for (int j = i + 1; j < listCropRegions.size(); j++) {
+                CropRegion otherRegion = listCropRegions.get(j);
+                assertFalse(
+                    otherRegion.getBoundingBox().getX() == region.getBoundingBox().getX()
+                    && otherRegion.getBoundingBox().getY() == region.getBoundingBox().getY()
+                    && otherRegion.getBoundingBox().getHeight() == region.getBoundingBox().getHeight()
+                    && otherRegion.getBoundingBox().getWidth() == region.getBoundingBox().getWidth());
+            }
+        }
+    }
+
+    private static void validateRead(ImageAnalysisResult result) {
+
+        ReadResult readResult = result.getRead();
+        assertNotNull(readResult);
+        assertNotNull(readResult.getBlocks());
+        assertEquals(readResult.getBlocks().size(), 1);
+
+        DetectedTextBlock block = readResult.getBlocks().get(0);
+        assertNotNull(block);
+
+        List<DetectedTextLine> lines = block.getLines();
+        assertNotNull(lines);
+        assertEquals(lines.size(), 3);
+
+        // Do some validations on the first line
+        DetectedTextLine line = lines.get(0);
+        assertNotNull(line);
+        assertTrue(line.getText().equals("Sample text"));
+
+        List<ImagePoint> polygon = line.getBoundingPolygon();
+        assertNotNull(polygon);
+        assertEquals(polygon.size(), 4);
+        for (int i = 0; i < polygon.size(); i++) {
+            assertTrue(polygon.get(i).getX() > 0);
+            assertTrue(polygon.get(i).getY() > 0);
+        }
+
+        // Do some verifications on the 3rd line
+        line = lines.get(2);
+        assertNotNull(line);
+        assertTrue(line.getText().equals("123 456"));
+
+        List<DetectedTextWord> words = line.getWords();
+        assertNotNull(words);
+        assertEquals(words.size(), 2);
+
+        DetectedTextWord word = words.get(1);
+        assertNotNull(word);
+        assertTrue(word.getText().equals("456"));
+        assertTrue(word.getConfidence() > 0.0);
+        assertTrue(word.getConfidence() < 1.0);
+
+        polygon = word.getBoundingPolygon();
+        assertNotNull(polygon);
+        assertEquals(polygon.size(), 4);
+        for (int i = 0; i < polygon.size(); i++) {
+            assertTrue(polygon.get(i).getX() > 0);
+            assertTrue(polygon.get(i).getY() > 0);
+        }
+    }
+
+    private static void printAnalysisResults(String testName, ImageAnalysisResult result) {
+
+        System.out.println(" ******************** TEST NAME: " + testName + " ******************** ");
```

Comment:
If it isn't required, I'd only run this if a test failed. Printing on all tests can make it harder to parse output from test runs to find the cause on failing tests (stack traces from failed tests, etc)

Reply:
@alzimmermsft I made a change to (by default) not print all the results. This is controlled by a final boolean variable in `ImageAnalysisClientTestBase.java`. So you can enable it in local runs if needed.